edelphi/src/main/typescript/src/components/panel/query-comment-list.tsx

```diff
@@ -1,86 +1,432 @@
+        oldestFirst: false
+      });
+
+      const combinedComments = [ ...comments ];
+      childComments.forEach(
+        childComment => {
+          const childIndex = combinedComments.findIndex(comment => comment.id === childComment.id);
+          if (childIndex > -1) {
+            combinedComments.splice(childIndex, 1, childComment);
+          } else {
+            combinedComments.push(childComment);
+          }
+        }
+      )
+
+      this.setState({
+        comments: combinedComments
+      });
+    } catch (error) {
+      console.error(error);
+    }
+  }
+
+  /**
+   * Handles query question comment notification MQTT message
+   *
+   * @param message message
+   */
+  private onQueryQuestionCommentNotification = async (message: QueryQuestionCommentNotification) => {
+    const { pageId, panelId } = this.props;
+
+    if (message.pageId !== pageId || message.panelId !== panelId) {
+      return;
+    }
+
+    const { commentId, parentCommentId, type } = message;
+
+    switch (type) {
+      case "CREATED":
+        this.onCreate(commentId);
+        break;
+      case "UPDATED":
+        this.onUpdate(commentId);
+        break;
+      case "DELETED":
+        this.onDelete(commentId, parentCommentId);
+        break;
+    }
+  }
+
+  /**
+   * Event handler for CREATED MQTT message
+   *
+   * @param commentId comment ID
+   */
+  private onCreate = async (commentId: number) => {
+    const { panelId, accessToken } = this.props;
+    const { comments } = this.state;
+
+    const commentList = [ ...comments ];
+
+    try {
+      const newComment = await Api.getQueryQuestionCommentsApi(accessToken).findQueryQuestionComment({
+        panelId: panelId,
+        commentId: commentId
+      });
+
+      if (newComment.parentId) {
+        const parentIndex = commentList.findIndex(comment => comment.id === newComment.parentId);
+
+        if (parentIndex > -1) {
+          const parent = commentList[parentIndex];
+          commentList.splice(parentIndex, 1, { ...parent, childCount: (parent.childCount || 0) + 1 });
+        }
+
+        this.setState({ comments: [ newComment, ...commentList ] }, () => this.loadComments(newComment.parentId!));
+
+      } else {
+        this.setState({ comments: [ newComment, ...commentList ] });
+      }
+    } catch (error) {
+      console.error(error);
+    }
+  }
+
+  /**
+   * Event handler for UPDATED MQTT message
+   *
+   * @param commentId comment ID
+   */
+  private onUpdate = async (commentId: number) => {
+    const { panelId, accessToken } = this.props;
+    const { comments } = this.state;
+
+    const commentList = [ ...comments ];
+    try {
+      const updatedComment = await Api.getQueryQuestionCommentsApi(accessToken).findQueryQuestionComment({
+        panelId: panelId,
+        commentId: commentId
+      });
+
+      const index = commentList.findIndex(comment => comment.id === updatedComment.id);
+
+      if (index < 0) {
+        commentList.push(updatedComment);
```

Comment:
Shouldn't selected order affect placement of the new comment (first or last) on the comment list?